src/IssueLabeler/AzureSdkIssueLabelerService.cs

```diff
@@ -14,40 +15,60 @@ namespace IssueLabeler
     {
+        private static readonly ActionResult EmptyResult = new JsonResult(new PredictionResponse(Array.Empty<string>()));
+
         public AzureSdkIssueLabelerService(ILabelerLite labeler, IModelHolderFactoryLite modelHolderFactory, IConfiguration config)
             : base(labeler, modelHolderFactory, config)
         {
         }
 
         [FunctionName("AzureSdkIssueLabelerService")]
-        public async Task<PredictionResponse> Run([HttpTrigger(AuthorizationLevel.Function, "POST", Route = null)] HttpRequest request, ILogger log)
+        public async Task<ActionResult> Run([HttpTrigger(AuthorizationLevel.Function, "POST", Route = null)] HttpRequest request, ILogger log)
         {
-            using var bodyReader = new StreamReader(request.Body);
-
-            var requestBody = await bodyReader.ReadToEndAsync();
-            var issue = JsonConvert.DeserializeObject<IssuePayload>(requestBody);
+            IssuePayload issue;
 
-            // In order for labels to be valid for Azure SDK use, there must
-            // be at least two of them, which corresponds to a Service (pink)
-            // and Category (yellow).  If that is not met, then no predictions
-            // should be returned.
-            var predictions = await Labeler.QueryLabelPrediction(
-                issue.IssueNumber,
-                issue.Title,
-                issue.Body,
-                issue.IssueUserLogin,
-                issue.RepositoryName,
-                issue.RepositoryOwnerName);
+            try
+            {
+                using var bodyReader = new StreamReader(request.Body);
 
-            if (predictions.Count < 2)
+                var requestBody = await bodyReader.ReadToEndAsync();
+                issue = JsonConvert.DeserializeObject<IssuePayload>(requestBody);
+            }
+            catch (Exception ex)
             {
-                return new PredictionResponse(Array.Empty<string>());
+                log.LogError($"Unable to deserialize payload:{ex.Message}{Environment.NewLine}\t{ex}{Environment.NewLine}");
+                return new BadRequestResult();
             }
 
-            return new PredictionResponse(predictions);
+            try
+            {
+                // In order for labels to be valid for Azure SDK use, there must
+                // be at least two of them, which corresponds to a Service (pink)
+                // and Category (yellow).  If that is not met, then no predictions
+                // should be returned.
+                var predictions = await Labeler.QueryLabelPrediction(
+                    issue.IssueNumber,
+                    issue.Title,
+                    issue.Body,
+                    issue.IssueUserLogin,
+                    issue.RepositoryName,
+                    issue.RepositoryOwnerName);
+
+                if (predictions.Count < 2)
+                {
+                    return EmptyResult;
+                }
```

Comment:
Would it be more appropriate to return some non-success error code here?

Reply:
If we were a public service, maybe.   In this case, our consumer is just the GH action bot.    Unless it made a mistake forming the payload, we just want it to see labels or no labels so that it can follow it's flow for initial triage.   If we decide that we want to go back to adding a comment with label candidates, we can extend the payload and tweak the bot logic - but we weren't doing anything with that information once we gained confidence in the predictions.    

For now, we'd like to keep things simple.